package/MDAnalysis/converters/OpenMMParser.py

```diff
@@ -105,42 +108,81 @@ def _mda_topology_from_omm_topology(self, omm_topology):
+             their names and their masses are then guessed using their atomtypes.
+           * When partial elements are present, values from available elements are used
+             whereas the absent elements are assigned an empty string with their
+             atomtype set to 'X' and mass set to 0.0.
+
+        """
         atom_resindex = [a.residue.index for a in omm_topology.atoms()]
         residue_segindex = [r.chain.index for r in omm_topology.residues()]
         atomids = [a.id for a in omm_topology.atoms()]
         atomnames = [a.name for a in omm_topology.atoms()]
         chainids = [a.residue.chain.id for a in omm_topology.atoms()]
-        elements = [a.element.symbol for a in omm_topology.atoms()]
-        atomtypes = [a.element.symbol for a in omm_topology.atoms()]
-        masses = [a.element.mass._value for a in omm_topology.atoms()]
         resnames = [r.name for r in omm_topology.residues()]
         resids = [r.index + 1 for r in omm_topology.residues()]
         resnums = resids.copy()
         segids = [str(c.index) for c in omm_topology.chains()]
         bonds = [(b.atom1.index, b.atom2.index) for b in omm_topology.bonds()]
         bond_orders = [b.order for b in omm_topology.bonds()]
         bond_types = [b.type for b in omm_topology.bonds()]
-
-        n_atoms = len(atomids)
-        n_residues = len(resids)
-        n_segments = len(segids)
+        elements = [None if a.element is None else a.element.symbol
+                    for a in omm_topology.atoms()]
 
         attrs = [
             Atomids(np.array(atomids, dtype=np.int32)),
             Atomnames(np.array(atomnames, dtype=object)),
-            Atomtypes(np.array(atomtypes, dtype=object)),
             Bonds(bonds, types=bond_types, order=bond_orders, guessed=False),
             ChainIDs(np.array(chainids, dtype=object)),
-            Elements(np.array(elements, dtype=object)),
-            Masses(np.array(masses, dtype=np.float32)),
             Resids(resids),
             Resnums(resnums),
             Resnames(resnames),
             Segids(segids),
         ]
 
+        if not any(elements):
+            atomtypes = guess_types(atomnames)
+            attrs.append(Atomtypes(np.array(atomtypes, dtype=object)))
+            masses = guess_masses(atomtypes)
+            attrs.append(Masses(np.array(masses)))
+            warnings.warn("Element information is missing, elements "
+                          "attribute will not be populated. "
+                          "Atomtype attribute will be guessed using atom "
+                          "name and mass will be guessed using atomtype."
+                          "See MDAnalysis.topology.guessers.")
+        else:
+            validated_elements = []
+            masses = []
+            for a in omm_topology.atoms():
+                elem = a.element
+                if elem is not None and elem.symbol.capitalize() in SYMB2Z:
```

Comment:
Do we have a case for thinking that OpenMM will ever had non-normal elements? (i.e. those that won't fit in SYMB2Z) the element attribute is not None? If not, then we don't need to check inside of SYMB2Z?

Reply:
I think you could.

```
>>> from openmm.app.element import Element
>>> Element(0, "*", "*", 0)
<Element *>
```

That being said, I would prefer not to lose any weird element information I had. I guess it would go into `Atomtype` and not `Element` if we want to ensure all our elements are valid.